api/app/models/event.rb

```diff
@@ -30,6 +30,14 @@ class Event < ApplicationRecord
   validates :ticket_price, numericality: { greater_than_or_equal_to: 0 }, allow_nil: false
   validates :participant_capacity, numericality: { greater_than_or_equal_to: 0 }, allow_nil: false
 
+  def joinable?
+    published? || published_unverified? || published_on_review? || published_rejected?
+  end
```

Comment:
joinable? includes published_rejected? among statuses that allow joining. Allowing users to join events that were rejected (presumably by review) is likely incorrect and could enable joining inappropriate events.

Confirm intended statuses. If only published events should be joinable, restrict accordingly:
```
def joinable?
  published? || published_unverified? || published_on_review?
end
```
Or make the set configurable / driven by a white-list constant and add unit tests that assert allowed statuses.